### PR TITLE
fix: replace "cloud & cognitive" with "Carbon for IBM Products" in docs

### PR DESCRIPTION
**Pull request by dcwarwick**
Contributes to #1422

Update various docs and similar, changing the branding from Cloud & Cognitive to Carbon for IBM Products. NB this PR does not attempt to do a complete job: references to the PAL can be updated when the PAL updates, and references to the library @carbon/ibm-cloud-cognitive can be updated once we are sure we are publishing @carbon/ibm-products in sync. This PR just updates mentions of the package in our docs, storybooks, and a few mentions in constants, ids, etc. This PR shouldn't change any actual use or functioning of the components.

#### How did you test and verify your work?

Ran tests and storybook.

### CODE REVIEW

**Comment by dcwarwick** [comment]
@andrea-gm FYI

File: packages/cloud-cognitive/src/components/Toolbar/_toolbar.scss [hunk 1]
```diff
@@ -8,7 +8,7 @@
 @import 'carbon-components/scss/components/button/button';
 @import '../../global/styles/project-settings';
 
-@include exports($name: cloud-cognitive--toolbar) {
+@include exports('toolbar') {
```
Comment (SimonFinney) [inline packages/cloud-cognitive/src/components/Toolbar/_toolbar.scss:11]: These styles will never be exported due to [naming conflicts with Carbon's toolbar](https://github.com/carbon-design-system/ibm-cloud-cognitive/blob/main/packages/cloud-cognitive/src/global/styles/_imported-carbon-modules.scss#L23).

```suggestion
@include exports('ibm-products--toolbar') {
```

**Review by SimonFinney — COMMENTED** [review]
I have one concern but otherwise this LGTM 🎉

**Review by davidmenendez — DISMISSED** [review]

File: packages/cloud-cognitive/src/components/Toolbar/_toolbar.scss (hunk 1 above)
Comment (dcwarwick) [reply packages/cloud-cognitive/src/components/Toolbar/_toolbar.scss]: Thanks, good catch

**Pull request merged**